internal/forms/forms.go

```diff
@@ -705,6 +709,108 @@ func (m *Manager) findAbsPathForTemplatePath(tmplPath string) (string, error) {
 	return retVal, nil
 }
 
+func (m *Manager) gpsPos() (gpsd.Position, error) {
+	// return current GPS Position
+
+	addr := m.config.GPSd.Addr
+	if addr != "" {
+		conn, err := gpsd.Dial(addr)
+		if err != nil {
+			log.Printf("GPSd daemon: %s", err)
+		}
+		defer conn.Close()
+
+		conn.Watch(true)
+
+		log.Println("Waiting for position from GPSd...")
+		pos, err := conn.NextPosTimeout(3 * time.Second) // TODO: configurable
+		if err != nil {
+			log.Printf("GPSd: %s", err)
+		} else {
+			log.Printf("GPSd: %f %f", pos.Lat, pos.Lon)
+		}
+		return pos, err
+	} else {
+		return gpsd.Position{}, errors.New("GPSd: not configured.")
+	}
+}
+
+type gpsStyle int
+
+const (
+	// documentation: https://www.winlink.org/sites/default/files/RMSE_FORMS/insertion_tags.zip
+	signedDecimal gpsStyle = iota // 41.1234 -73.4567
+	decimal                       // 46.3795N 121.5835W
+	degreeMinute                  // 46-22.77N 121-35.01W
+)
+
+func gpsFmt(style gpsStyle, pos gpsd.Position) string {
+	var northing string
+	var easting string
+	var latDegrees int
+	var latMinutes float64
+	var lonDegrees int
+	var lonMinutes float64
+
+	noPos := gpsd.Position{}
+	if pos == noPos {
+		return "(Not available)"
+	}
+	switch style {
+	case degreeMinute:
+		{
+			latDegrees = int(math.Trunc(math.Abs(pos.Lat)))
+			latMinutes = (math.Abs(pos.Lat) - float64(latDegrees)) * 60
+			lonDegrees = int(math.Trunc(math.Abs(pos.Lon)))
+			lonMinutes = (math.Abs(pos.Lon) - float64(lonDegrees)) * 60
+		}
+		fallthrough
+	case decimal:
+		{
+			if pos.Lat >= 0 {
+				northing = "N"
+			} else {
+				northing = "S"
+			}
+			if pos.Lon >= 0 {
+				easting = "E"
+			} else {
+				easting = "W"
+			}
+		}
+	}
+
+	switch style {
+	case signedDecimal:
+		return fmt.Sprintf("%.4f %.4f", pos.Lat, pos.Lon)
+	case decimal:
+		return fmt.Sprintf("%.4f%s %.4f%s", math.Abs(pos.Lat), northing, math.Abs(pos.Lon), easting)
+	case degreeMinute:
+		return fmt.Sprintf("%02d-%05.2f%s %03d-%05.2f%s", latDegrees, latMinutes, northing, lonDegrees, lonMinutes, easting)
+	default:
+		return "(Not available)"
+	}
+}
+
+func posToGridSquare(pos gpsd.Position) string {
```

Comment:
We're already depending on github.com/pd0mz/go-maidenhead for doing maidenhead calculation. Maybe it provides some of this functionality already?

Reply:
Well, yes, there is! Thanks for that.